File: 1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.cpp
class Solution {
public:
    string generate(int n, char start) {
        string res = "";
        char ch = start;
        for (int i = 0; i < n; i++) {
            res += ch;
            if (res.back() == '0')
                ch = '1';
            else
                ch = '0';
        }
        return res;
    }
    int minOperations(string s) {
        int n = s.length(), minOps = INT_MAX;
        string startZero = generate(n, '0');
        string startOne = generate(n, '1');
        int cnt0 = 0, cnt1 = 0;
        for (int i = 0; i < n; i++) {
            if (s[i] == startZero[i])
                continue;
            else
                cnt0++;
        }
        for (int i = 0; i < n; i++) {
            if (s[i] == startOne[i])
                continue;
            else
                cnt1++;
        }
        minOps = min(cnt0, cnt1);
        return minOps;
    }
};

File: 1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.py
class Solution:
    def minOperations(self, s: str) -> int:
        minOps = 1e9 + 7
        n = len(s)
        cnt1 = cnt2 = 0
        for i in range(n):
            expectOne = "0" if (i & 1) else "1"
            expectZero = "1" if (i & 1) else "0"
            if s[i] != expectOne:
                cnt1 = cnt1 + 1
            if s[i] != expectZero:
                cnt2 = cnt2 + 1
        minOps = min(cnt1, cnt2)
        return minOps
